webviz_subsurface/_utils/vector_calculator.py

```diff
@@ -0,0 +1,325 @@
+) -> List[VariableVectorMapInfo]:
+    variable_vector_map: List[VariableVectorMapInfo] = []
+    for variable in variable_vector_dict:
+        variable_vector_map.append(
+            {
+                "variableName": variable,
+                "vectorName": [variable_vector_dict[variable]],
+            }
+        )
+    return variable_vector_map
+
+
+def expressions_from_config(
+    predefined_expressions: Optional[Path],
+) -> List[ExpressionInfo]:
+    output: List[ExpressionInfo] = []
+
+    if predefined_expressions is None:
+        return output
+
+    expressions: Dict[str, ConfigExpressionData] = yaml.safe_load(
+        predefined_expressions.read_text()
+    )
+
+    for expression in expressions:
+        expression_dict = {
+            "name": expression,
+            "expression": expressions[expression]["expression"],
+            "id": f"{uuid4()}",
+            "variableVectorMap": variable_vector_map_from_dict(
+                expressions[expression]["variableVectorMap"]
+            ),
+            "isValid": False,  # Set False and validate in seperate operation
+            "isDeletable": False,
+        }
+        if "description" in expressions[expression].keys():
+            expression_dict["description"] = expressions[expression]["description"]
+
+        output.append(expression_dict)
+    return output
+
+
+def is_vector_name_existing(name: str, vector_data: list) -> bool:
+    nodes = name.split(":")
+    current_child_list = vector_data
+    for node in nodes:
+        found = False
+        for child in current_child_list:
+            if child["name"] == node:
+                children = child["children"]
+                current_child_list = children if children is not None else []
+                found = True
+                break
+        if not found:
+            return False
+    return found
+
+
+def get_expression_from_name(
+    name: str, expressions: List[ExpressionInfo]
+) -> Union[ExpressionInfo, None]:
+    for expr in expressions:
+        if name == expr["name"]:
+            return expr
+    return None
+
+
+def get_custom_vector_definitions_from_expressions(
+    expressions: List[ExpressionInfo],
+) -> dict:
+    """
+    Get custom vector definitions for vector selector from list of calculated expressions.
+
+    VectorSelector has VectorDefinitions which is utilized for calculated expressions.
+
+    `VectorDefinitions:`
+        key: str, vector name
+        value: {
+            type: str, defined vector type
+            description: str, description of vector
+        }
+
+    `Note:`
+    Uses expression str as description if optional expression description str does not exist.
+    """
+
+    output: dict = {}
+    for expression in expressions:
+        name = expression["name"]
+        key = name.split(":")[0]
+        vector_type = "calculated"
+        description = (
+            expression["expression"]
+            if not "description" in dict(expression).keys()
```

Comment:
Isn't `expression` already a dict?
```suggestion
            if not "description" in expression
```